src/chainlit/frontend/src/components/atoms/switch.tsx

```diff
@@ -3,27 +3,48 @@ import { green } from 'palette';
 import MSwitch, { SwitchProps as MSwitchProps } from '@mui/material/Switch';
 import { styled } from '@mui/material/styles';
 
-type SwitchProps = {
+import InputStateHandler from 'components/organisms/inputs/inputStateHandler';
+
+import { IInput } from 'types/Input';
+
+export type SwitchProps = {
   checked?: boolean;
-  disabled?: boolean;
   inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
   onChange: (
     event?: React.ChangeEvent<HTMLInputElement>,
     checked?: boolean
   ) => void;
-};
+} & IInput;
 
 export default function Switch(props: SwitchProps): JSX.Element {
-  const { checked, disabled, inputProps, onChange } = props;
+  const {
+    checked,
+    description,
+    disabled,
+    hasError,
+    id,
```

Comment:
I noticed that some current Switch implementations now have errors since id is required